src/Microsoft.Framework.Configuration.Abstractions/IConfigurationRoot.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Framework.Configuration
+{
+    public interface IConfigurationRoot : IConfiguration
+    {
+        void Reload();
```

Comment:
Off topic but this just made me wonder if it is completely intentional that we don't have async APIs in Configuration. I imagine @lodejard will have the scoop on that.